datastore/src/main/scala/org/allenai/datastore/Datastore.scala

```diff
@@ -162,6 +80,16 @@ class Datastore(val name: String, val s3: AmazonS3Client) extends Logging {
     cause
   )
 
+  /** Exception indicating that the locator provided was for the wrong datastore. This is thrown if
+    * the "datastore" field in a locator doesn't match this datastore's name.
+    * @param locator the bad locator
+    */
+  class BadLocatorException(locator: Locator)
+    extends DsException(
+      s"locator for ${locator.datastore} doesn't match this datastore ($name)",
+      null
+    )
```

Comment:
I like the idea of having just one `Datastore`. In this situation, all of the multi-argument `path` / `publish` / `exists` / `url` methods would require a datastore name, which seems totally fine to me.

This would definitely justify a 2.0 update . . .

Sound reasonable?